src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java

```diff
@@ -71,20 +89,28 @@ public void delete(Long id) {
-        save(new Product(5L, "iphone6", "Apple iPhone 6", new BigDecimal(1000), usd, 30, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Apple/Apple%20iPhone%206.jpg"));
-        save(new Product(6L, "htces4g", "HTC EVO Shift 4G", new BigDecimal(320), usd, 3, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/HTC/HTC%20EVO%20Shift%204G.jpg"));
-        save(new Product(7L, "sec901", "Sony Ericsson C901", new BigDecimal(420), usd, 30, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Sony/Sony%20Ericsson%20C901.jpg"));
-        save(new Product(8L, "xperiaxz", "Sony Xperia XZ", new BigDecimal(120), usd, 100, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Sony/Sony%20Xperia%20XZ.jpg"));
-        save(new Product(9L, "nokia3310", "Nokia 3310", new BigDecimal(70), usd, 100, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Nokia/Nokia%203310.jpg"));
-        save(new Product(10L, "palmp", "Palm Pixi", new BigDecimal(170), usd, 30, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Palm/Palm%20Pixi.jpg"));
-        save(new Product(11L, "simc56", "Siemens C56", new BigDecimal(70), usd, 20, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Siemens/Siemens%20C56.jpg"));
-        save(new Product(12L, "simc61", "Siemens C61", new BigDecimal(80), usd, 30, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Siemens/Siemens%20C61.jpg"));
-        save(new Product(13L, "simsxg75", "Siemens SXG75", new BigDecimal(150), usd, 40, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Siemens/Siemens%20SXG75.jpg"));
+    private boolean isProductSuitableForQuery(Product product, String query) {
+        if (query == null || query.isEmpty()) {
+            return true;
+        }
+        return numberOfMatchesWithQuery(product, query) != 0;
+    }
+
+    private int coincidences(Product product, String query) {
+        if (query == null || query.isEmpty()) {
+            return 1;//default val
+        }
+        int productDescriptionWordsNumber = product.getDescription().split(" ").length;
+        return  numberOfMatchesWithQuery(product,query) * 100 / productDescriptionWordsNumber;
+    }
+
+    private int numberOfMatchesWithQuery(Product product, String query) {
+        List<String> productDescriptions = Arrays.asList(product.getDescription().toLowerCase().split(" "));
+        List<String> queries = Arrays.asList(query.toLowerCase().split(" "));
+        List<Boolean> matches = queries.stream()
+                .map(productDescriptions::contains)
+                .filter(aBoolean -> aBoolean)
```

Comment:
Why do we need this?

`filter(aBoolean -> aBoolean)`

What is aBoolean here?

Reply:
This is for creating a List that consists only of true elements, so the List size is equals to the number of matches